File: y2023_Java/src/general/Day.java
package general;

import java.util.Objects;

public abstract class Day {
    private final String INPUT_FILENAME = "input.txt";
    protected String[] input;
    protected int resultPartOne;
    protected int resultPartTwo;

    public Day() {
        String inputPath = getInputPath();
        input = new FileInput(inputPath).getFileInput();
        resultPartOne = 0;
        resultPartTwo = 0;
    }

    protected String getInputPath() {
        return Objects.requireNonNull(getClass().getResource(INPUT_FILENAME)).getPath();
    }
    public abstract int partOne();
    public abstract int partTwo();
}

File: y2024_C#/AdventOfCode/_General/Day.cs
namespace AdventofCode._General;

public abstract class Day
{
    private const string InputDirectory = "/home/korbinian/data/code/advent-of-code/y2024_C#/AdventOfCode/_Inputs";
    protected readonly List<string> Input;
    protected int ResultPartOne;
    protected int ResultPartTwo;

    protected Day(string inputFileName)
    {
        var inputPath = Path.Combine(InputDirectory, inputFileName);
        var inputArray = File.ReadAllLines(inputPath)
            .Select(line => line.Trim())
            .ToArray();
        Input = inputArray.ToList();
    }


    public abstract int PartOne();
    public abstract int PartTwo();
}